src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -359,27 +434,34 @@ function calc_volume_integral!(du, u, volume_integral, mesh,
   @unpack entropy_projected_u_values, Ph, sparsity_pattern = cache
   @unpack fluxdiff_local_threaded, rhs_local_threaded = cache
   rd = dg.basis
-  skip_index(i,j) = i > rd.Nq && j > rd.Nq
 
   flux_conservative, flux_nonconservative = volume_integral.volume_flux
 
   @threaded for e in eachelement(mesh, dg, cache)
     fluxdiff_local = fluxdiff_local_threaded[Threads.threadid()]
     fill!(fluxdiff_local, zero(eltype(fluxdiff_local)))
     u_local = view(entropy_projected_u_values, :, e)
-    for i in eachdim(mesh)
-      # Todo: DGMulti. Dispatch on curved meshes, this code only works for affine meshes (accessing rxJ[1,e],...)
-      Qi_skew = build_lazy_physical_derivative(e, i, mesh, dg, cache)
 
-      # compute conservative flux differencing contribution
-      hadamard_sum!(fluxdiff_local, Qi_skew, flux_conservative, i,
-                    u_local, equations, sparsity_pattern, skip_index)
+    # Val{true}() indicates flux is symmetric
+    local_flux_differencing!(fluxdiff_local, u_local, e,
+                             Val{true}(), flux_conservative,
+                             has_sparse_operators(dg),
+                             mesh, equations, dg, cache)
+
+    # Compute non-conservative contributions. Non-conservative fluxes for MHD don't support
+    # `orientation::AbstractVector`, so we use numerical fluxes in Cartesian directions
+    # and sum up the discrete derivative operators per coordinate direction accordingly.
+    for dim in eachdim(mesh)
+      Q_skew = build_lazy_physical_derivative(e, dim, mesh, dg, cache)
 
       # Scale the non-conservative part (it doesn't include 1/2 factor for a central flux).
       # This effectively removes the multiplication by two included in `Qi_skew`.
-      half_Qi_skew = LazyMatrixLinearCombo(tuple(Qi_skew), tuple(0.5))
-      hadamard_sum_nonsymmetric!(fluxdiff_local, half_Qi_skew, flux_nonconservative, i,
-                                 u_local, equations, sparsity_pattern, skip_index)
+      half_Q_skew = LazyMatrixLinearCombo(tuple(Q_skew), tuple(0.5))
```

Comment:
```suggestion
      half_Q_skew = LazyMatrixLinearCombo((Q_skew, ), (0.5, ))
```
Just a matter of taste but that's the version you can read in the REPL output. Feel free to silently ignore and resolve this.